### ISSUE TITLE
Unit test failure: MultiLayerTest.testBatchNorm

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#1481]
Looks like batch norm (or, perhaps only this test - not sure) is currently broken:

BatchNormalizationParamInitializer.createGamma does a layer.getNOut(), but in that test (and perhaps in general) the batch norm layer has nOut = 0

See also:
BatchNormalizationTest.testCnnShapeBatchNormBack and .testDnnShapeBatchNormBack

cc @nyghtowl
